### PR TITLE
[Fix #997] Careers Static Terraform

### LINKED ISSUE #997: S3 bucket automation for careers
See https://github.com/mozmeao/infra/pull/669 for example

### PR DESCRIPTION
**Pull request by glogiotatidis**
Creates
 - prod bucket
 - staging bucket
 - logs bucket
 - IAM user with S3 permissions to the buckets above
 - CF distribution for staging
 - CF distribution for prod

I need help getting certificates for the CF distributions. We have certificates for careers but they are both on us-west-2.

The above has been applied. You can test the staging CF dist at https://dmin398ukjj8z.cloudfront.net/

### CODE REVIEW

**Comment by glogiotatidis** [comment]
Marked as WIP, to update with the needed certs.

File: apps/careers/tf/careers.tf [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+terraform {
+  backend "s3" {
+    bucket = "mozilla-careers-s3-provisioning-tf-state"
+    key    = "tf-state"
+    region = "us-west-2"
+  }
+}
+
+resource "aws_iam_user" "careers" {
```
Comment (metadave) [inline apps/careers/tf/careers.tf]: unless it needs to be public, we store all IAM TF [here](https://github.com/mozmeao/infra-private/tree/master/aws/iam_policies)

**Comment by metadave** [comment]
@glogiotatidis what names would you like on the certs?

File: apps/careers/tf/careers.tf (hunk 1 above)
Comment (glogiotatidis) [reply apps/careers/tf/careers.tf]: Removed from here and opened a PR against private repo. https://github.com/mozmeao/infra-private/pull/153

Thanks!

**Comment by glogiotatidis** [comment]
> @glogiotatidis what names would you like on the certs?

careers-prod.moz.works
careers.mozilla.org 

and

careers-stage.moz.works 
careers.allizom.org 

Thanks!

**Comment by metadave** [comment]
@jgmize can you review the CloudFront `aliases` values?

**Comment by glogiotatidis** [comment]
Updated with cert arns.

**Comment by metadave** [comment]
@jgmize can you review the CloudFront aliases values?

**Pull request merged**